readthedocs/oauth/services/github.py

```diff
@@ -203,7 +203,7 @@ def setup_webhook(self, project, integration=None):
         """
         session = self.get_session()
         owner, repo = build_utils.get_github_username_repo(url=project.repo)
-        if integration:
+        if integration and not integration.secret:
```

Comment:
@ericholscher When we pass the integration from `update_webhook` to `setup_webhook` on `404`the integration already has a secret so this condition always returns `False`  and creates a new integration. thats why some tests are failing.